tests/freshness.rs

```diff
@@ -172,16 +172,192 @@ fn changing_features_is_ok() {
 [FINISHED] debug [unoptimized + debuginfo] target(s) in [..]
 "));
 
+    /* Targets should be cached from the first build */
+
+    assert_that(p.cargo("build"),
+                execs().with_status(0)
+                       .with_stderr("\
+[FINISHED] debug [unoptimized + debuginfo] target(s) in [..]
+"));
+
     assert_that(p.cargo("build"),
                 execs().with_status(0)
+                       .with_stdout(""));
+
+    assert_that(p.cargo("build").arg("--features").arg("foo"),
+                execs().with_status(0)
+                       .with_stderr("\
+[FINISHED] debug [unoptimized + debuginfo] target(s) in [..]
+"));
+}
+
+#[test]
+fn changing_bin_paths_common_target_features_caches_targets() {
+    let p = project("foo")
+        .file(".cargo/config", r#"
+            [build]
+            target-dir = "./target"
+        "#)
+        .file("dep_crate/Cargo.toml", r#"
+            [package]
+            name    = "dep_crate"
+            version = "0.0.1"
+            authors = []
+
+            [features]
+            ftest  = []
+        "#)
+        .file("dep_crate/src/lib.rs", r#"
+            #[cfg(feature = "ftest")]
```

Comment:
I would expect this test to fail because I conditionally compile this dep_crate to do different things on each branch depending on feature flags (and they're definitely building to one lib file with no suffixes)

```
nipunn-mbp:foo nipunn$ ls target/debug/
a         a.dSYM    b         b.dSYM    build     deps      examples  liba.rlib libb.rlib native
nipunn-mbp:foo nipunn$ ls target/debug/deps/
liba.rlib         libb.rlib         libdep_crate.rlib
```